### PR TITLE
Turret mounts

### LINKED ISSUE #19206: Mounted Machine Guns Should Not Require Cutting Tools
Most machine gun mounts will utilize a quick release pin system of some sort. 

Please review the following. 

https://www.armyproperty.com/Resources/PS-Mag/2004-08/Vehicular%20Machine-Gun%20Mounting%20Systems.pdf

Search for M197 in the following document. 

http://www.liberatedmanuals.com/TM-9-1005-224-10.pdf

### PR DESCRIPTION
**Pull request by Night-Pryanik**
- Added `turret mount` as an item and as a vehicle part;
- All turrets (vehicle parts with `TURRET` flag) can be installed only on turret mount. Hopefully this will bring us closer to the time when we'll able to mount arbitrary guns on this turret mount;
- Turret mount are installed in around 2 hours, turrets are installed on mount in 10 minutes;
- Turrets can be removed from mounts without any tools, turret mounts can be removed with standard tools (hacksaw and the like);
- All vehicle templates in vanilla and in `Tanks` mod with pre-installed turrets are updated by adding turret mounts;
- All vehicles in older games or with mods that add turrets automatically gain turret mounts.
- Closes #19206.

### CODE REVIEW

**Comment by BorkBorkGoesTheCode** [comment]
savegame.json.cpp has code that handles migrating vehicle parts, I'm looking through it for a way to place turret mounts under turrets. https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/savegame_json.cpp

File: src/vehicle.cpp [hunk 1]
```diff
@@ -1648,6 +1648,19 @@ bool vehicle::can_mount(int const dx, int const dy, const vpart_id &id) const
         }
     }
 
+    //Turrets must be installed on a turret mount
+    if( part.has_flag( "TURRET" ) ) {
+        bool anchor_found = false;
+        for( const auto &elem : parts_in_square ) {
+            if( part_info( elem ).has_flag( "TURRET_MOUNT" ) ) {
+                anchor_found = true;
```
Comment (woutershep) [inline src/vehicle.cpp]: break;

Comment (Night-Pryanik) [reply src/vehicle.cpp]: There is no `break;`s in similar snippets above and below here in `vehicle.cpp`.

Comment (woutershep) [reply src/vehicle.cpp]: Well in this specific case its pretty clear that once an anchor is found additional parts don't need to be checked for the TURRET_MOUNT flag. The only possible scenario after setting anchor_found to true is setting anchor_mount to true again, which it already is, so a no-op. It's more efficient to stop checking than to do further processing that can't change anything.

Other parts in the file do it properly, just to pick a few:
https://github.com/CleverRaven/Cataclysm-DDA/blob/bb912354e10d7e8ea510de335b6572100ec2634a/src/vehicle.cpp#L206
https://github.com/CleverRaven/Cataclysm-DDA/blob/bb912354e10d7e8ea510de335b6572100ec2634a/src/vehicle.cpp#L731
https://github.com/CleverRaven/Cataclysm-DDA/blob/bb912354e10d7e8ea510de335b6572100ec2634a/src/vehicle.cpp#L5380

File: data/json/items/vehicle_parts.json [hunk 2]
```diff
@@ -1018,5 +1018,17 @@
         "weight": 500,
         "volume": 1,
         "price": 800
+    },{
+        "id": "turret_mount",
+        "type":"GENERIC",
+        "name": "turret mount",
+        "description": "A universal mount for weapons intended to be installed as a turrets.",
```
Comment (BrettDong) [inline data/json/items/vehicle_parts.json]: > a turrets

a turret?

**Comment by codemime** [comment]
Still breaks a unit test (presumably due to the heavier vehicle).

**Comment by Night-Pryanik** [comment]
Yeah, I know. I'm trying to run tests for my first time.
I'm trying to do `make` in `tests` dir, but it is showing me `No target to rule ../cataclysm.a'`. Am I doing it wrong?

**Comment by kevingranade** [comment]
Run ```make check``` from the main directory.

**Comment by Night-Pryanik** [comment]
```
g++  -DGIT_VERSION -DLOCALIZE -ffast-math -O0 -Wall -Wextra -D_GLIBCXX_DEBUG -g  -fsigned-char -std=c++11 -MMD -MP -c src/action.cpp -o obj/action.o
process_begin: CreateProcess(NULL, g++ -DGIT_VERSION -DLOCALIZE -ffast-math -O0 -Wall -Wextra -D_GLIBCXX_DEBUG -g -fsigned-char -std=c++11 -MMD -MP -c src/action.cpp -o obj/action.o, ...) failed.
make (e=2): the system cannot find the file specified.
make: *** [obj/action.o] Error 2
```

File: src/savegame_json.cpp [hunk 3]
```diff
@@ -1913,6 +1913,11 @@ void vehicle::deserialize(JsonIn &jsin)
             }
         }
     }
+
+    for( auto turret : get_parts( "TURRET", false ) ) {
+        install_part( turret->mount.x, turret->mount.y, static_cast<vpart_id>( "turret_mount" ), false );
```
Comment (BevapDin) [inline src/savegame_json.cpp]: This is a unusual way of creating a `vpart_id` instance. Normally one just creates a temporary object directly: `vpart_id( "the-id" )`.

**Comment by illi-kun** [comment]
Any reason to keep '[CR]' tag?

**Comment by ZhilkinSerg** [comment]
There are some issues with three vehicles.

# Enabled mods:

image

# Affected vehicles with turrets:

image

image

image

**Comment by smolbird** [comment]
What about manual mounted weapons? How are they affected? And tankmod?

**Pull request merged**